POEM_052.md

```diff
@@ -16,6 +16,112 @@ Associated implementation PR:
+
+
+For model design where the code is being written for an OpenMDAO component, some coupling is acceptable to streamline workflow (reduce repeat code). For example, XDSM to spec test work flow.
+
+```python
+def some_component_func(x, y, z) -> ['foo', 'bar']:
+    return
+
+SomeComponent = om.ExplicitFuncComp(some_component_func)
+```
+
+The resulting `SomeComponent` has inputs and outputs with the least restrictive metadata possible and can be placed in a group so that an XDSM may be drawn and spec test regression data may be generated.
+This is an important workflow because it reduces duplicate code maintenance. If I/O needs to change, the component can be changed, the XDSM and spec files re-generated and reviewed, then regression data can be updated. This makes user code "DRY" -- "do not repeat yourself", an important coding principle.
+Does OpenMDAO run the compute to test things during setup? In other words, does this skeleton  need to return something?
+
+Next steps include writing the function code and configuring the Component.
+Component configuration is what's included in the I/O and "everything else".
+For the I/O, I will ignore the fully de-coupled engineering code/ OpenMDAO API case.
+For the coupled case, we can do more annotations or decorators.
+
+For the annotations, it could be nice to provide a custom type that takes the metadata fields as var. And read defaults as default value like:
+
+```python
+def some_component_func(x: om.func.Input(units='m') = 0., y, z) ->
+                        [om.func.Output('foo'),
+                         om.func.Output('bar', units='1/m', shape=4)]:
+    return
+
+SomeComponent = om.ExplicitFuncComp(some_component_func)
+```
+Or if it's better to use a dict or 3rd party metadata container, that's fine too. Or support both. 
+
+Or, the input/output can be configured by decorator as proposed by Justin. I would like it if the input decorators inspected the name of the input variable by default. 
+
+For more involved OpenMDAO API calls, we can use a subclass to access the the full API. See below.
+Another feature is other functions having a normal signature, either as stand-alone functions or methods of subclass. Could use standard OpenMDAO method name and decorate to handle function input/output. For standalone, may need special API call or flag.
+Or are both cases solved by just assuming that signature type for this subclass and wrapping everything with an IO handler.
+
+```python
+class SomeComponent(om.ExplicitFuncComp(some_component_func)):
+    def setup(self):
+        super().setup()
+        """
+        can make any any component api calls here. If desired, could provide hooks (e.g., pre/post setup)
+        could do something similar with initialize, or any other method.
+
+        Is it possible to manipulate a 
+        """
+
+     @om.func.process_io # or maybe this is the default behavior for this class?
```

Comment:
This is super awkward. You have a class, and a method that looks like it should use the same API as the class (because it has the same name). But it *does not* use the same API! 

That will be epicly confusing. I really dislike this approach.